core/src/main/java/org/apache/iceberg/BaseIncrementalScan.java

```diff
@@ -45,6 +66,24 @@ public ThisT fromSnapshotInclusive(long fromSnapshotId) {
     return newRefinedScan(tableOps(), table(), schema(), newContext);
   }
 
+  @Override
+  public ThisT fromSnapshotExclusive(long fromSnapshotId, String referenceName) {
+    // for exclusive behavior, table().snapshot(fromSnapshotId) check can't be applied
+    // as fromSnapshotId could be matched to a parent snapshot that is already expired
+    SnapshotRef snapshotRef = table().refs().get(referenceName);
+    Preconditions.checkArgument(snapshotRef != null, "Cannot find ref %s", referenceName);
+    Preconditions.checkArgument(snapshotRef.isBranch(), "Ref %s is not a branch", referenceName);
```

Comment:
just curious, why wouldn't this work with tags?

Reply:
From my personal understanding, tag is just a snapshot point. It does not have a linear snapshot history, which is necessary for incremental reading.